Gems/Atom/RHI/Code/Tests/MultiDevicePipelineStateTests.cpp

```diff
@@ -0,0 +1,378 @@
+
+            ScrambleMemory(&desc.m_renderStates, sizeof(RHI::RenderStates), randomSeed++);
+
+            desc.m_renderStates.m_depthStencilState.m_depth.m_enable = true;
+            auto& renderAttachmentLayout = desc.m_renderAttachmentConfiguration.m_renderAttachmentLayout;
+            renderAttachmentLayout.m_attachmentCount = 2;
+            renderAttachmentLayout.m_attachmentFormats[0] = RHI::Format::R32_FLOAT;
+            renderAttachmentLayout.m_attachmentFormats[1] = RHI::Format::R8G8B8A8_SNORM;
+            renderAttachmentLayout.m_subpassCount = 1;
+            renderAttachmentLayout.m_subpassLayouts[0].m_rendertargetCount = 1;
+            renderAttachmentLayout.m_subpassLayouts[0].m_rendertargetDescriptors[0] =
+                RHI::RenderAttachmentDescriptor{ 1, RHI::InvalidRenderAttachmentIndex, RHI::AttachmentLoadStoreAction() };
+            renderAttachmentLayout.m_subpassLayouts[0].m_depthStencilDescriptor =
+                RHI::RenderAttachmentDescriptor{ 0, RHI::InvalidRenderAttachmentIndex, RHI::AttachmentLoadStoreAction() };
+            desc.m_renderAttachmentConfiguration.m_subpassIndex = 0;
+            return desc;
+        }
+
+        void ValidateCacheIntegrity([[maybe_unused]] RHI::Ptr<RHI::PipelineStateCache>& cache) const
+        {
+            EXPECT_TRUE(false);
+            /*
+            TODO: Once cache becomes multi-device, add that back in and also add MultiDevicePipelienStateTests as friend class to
+            PipelineStateCache
+             */
+            // cache->ValidateCacheIntegrity();
+        }
+
+    private:
+        void SetUp() override
+        {
+            MultiDeviceRHITestFixture::SetUp();
+
+            m_pipelineLayout = RHI::PipelineLayoutDescriptor::Create();
+            m_pipelineLayout->Finalize();
+        }
+
+        void TearDown() override
+        {
+            m_pipelineLayout = nullptr;
+            MultiDeviceRHITestFixture::TearDown();
+        }
+
+        RHI::Ptr<RHI::PipelineLayoutDescriptor> m_pipelineLayout;
+    };
+
+    TEST_F(MultiDevicePipelineStateTests, PipelineState_CreateEmpty_Test)
+    {
+        RHI::Ptr<RHI::MultiDevicePipelineState> empty = aznew RHI::MultiDevicePipelineState;
+        EXPECT_NE(empty.get(), nullptr);
+        EXPECT_FALSE(empty->IsInitialized());
+    }
+
+    TEST_F(MultiDevicePipelineStateTests, PipelineState_Init_Test)
+    {
+        RHI::Ptr<RHI::MultiDevicePipelineState> pipelineState = aznew RHI::MultiDevicePipelineState;
+        RHI::ResultCode resultCode = pipelineState->Init(DeviceMask, CreatePipelineStateDescriptor(0));
+        EXPECT_EQ(resultCode, RHI::ResultCode::Success);
+
+        // Second init should fail and throw validation error.
+        AZ_TEST_START_ASSERTTEST;
+        resultCode = pipelineState->Init(DeviceMask, CreatePipelineStateDescriptor(0));
+        AZ_TEST_STOP_ASSERTTEST(1);
+
+        EXPECT_EQ(resultCode, RHI::ResultCode::InvalidOperation);
```

Comment:
These all seem to be copy pasted with MD version. We really should be adding new tests related to testing single device PSOs. Given that the single device will be running these tests anyways I would focus on the multi device functionality of the new tests. Think of the new code added in the MD version and just focus unit testing around that code.